src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -104,18 +103,22 @@ trait RegularSync extends BlockBroadcast {
         val blockNumber = appStateStorage.getBestBlockNumber()
         val request = GetBlockHeaders(Left(blockNumber + 1), blockHeadersPerRequest, skip = 0, reverse = false)
         waitingForActor = Some(context.actorOf(
-          SyncBlockHeadersRequestHandler.props(peer, etcPeerManager, peerEventBus, request, resolveBranches = false)))
+          SyncBlockHeadersRequestHandler.props(peer, peerResponseTimeout, etcPeerManager, peerEventBus, request, resolveBranches = false)))
       case None =>
         log.debug("No peers to download from")
         scheduleResume()
     }
   }
 
   private def handleBlockBranchResolution(peer: Peer, message: Seq[BlockHeader]) =
-    //todo limit max branch depth? [EC-248]
     if (message.nonEmpty && message.last.hash == headersQueue.head.parentHash) {
       headersQueue = message ++ headersQueue
-      processBlockHeaders(peer, headersQueue)
+      if (headersQueue.length > branchResolutionMaxDepth) {
```

Comment:
yes, you are right it will fail in that case, this should be:
((headersQueue.length - 1) > branchResolutionMaxDepth), and if it is equal it should go to `processBlockHeaders()` and check if branch is resolved